fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraLdpIT.java

```diff
@@ -708,6 +708,128 @@ public void testPutBinary() throws IOException {
+        final String descLocation;
+
+        try (final CloseableHttpResponse response = execute(method)) {
+            binaryEtag1 = response.getFirstHeader("ETag").getValue();
+            binaryLastModed1 = response.getFirstHeader("Last-Modified").getValue();
+            descLocation = Link.valueOf(response.getFirstHeader("Link").getValue()).getUri().toString();
+        }
+
+        // First check ETags and Last-Modified headers for the binary
+        final HttpGet get1 = new HttpGet(location);
+        get1.addHeader("If-None-Match", binaryEtag1);
+        assertEquals("Expected 304 Not Modified", NOT_MODIFIED.getStatusCode(), getStatus(get1));
+
+        final HttpGet get2 = new HttpGet(location);
+        get2.addHeader("If-Modified-Since", binaryLastModed1);
+        assertEquals("Expected 304 Not Modified", NOT_MODIFIED.getStatusCode(), getStatus(get2));
+
+        // Next, check ETags and Last-Modified headers on the description
+        final HttpGet get3 = new HttpGet(descLocation);
+        try (final CloseableHttpResponse response = execute(get3)) {
+            descEtag1 = response.getFirstHeader("ETag").getValue();
+            descLastModed1 = response.getFirstHeader("Last-Modified").getValue();
+        }
+        assertNotEquals("Binary, description ETags should be different", binaryEtag1, descEtag1);
+
+        final HttpGet get4 = new HttpGet(descLocation);
+        get4.addHeader("If-None-Match", descEtag1);
+        assertEquals("Expected 304 Not Modified", NOT_MODIFIED.getStatusCode(), getStatus(get4));
+
+        final HttpGet get5 = new HttpGet(descLocation);
+        get5.addHeader("If-Modified-Since", descLastModed1);
+        assertEquals("Expected 304 Not Modified", NOT_MODIFIED.getStatusCode(), getStatus(get5));
+
+        // Pause two seconds before updating the description
+        sleep(2000);
+
+        // Next, update the description
+        final HttpPatch httpPatch = patchObjMethod(id + "/binary/fcr:metadata");
+        httpPatch.addHeader("Content-Type", "application/sparql-update");
+        httpPatch.setEntity(new StringEntity(
+                "INSERT { <> <http://purl.org/dc/elements/1.1/title> 'this is a title' } WHERE {}"));
+        assertEquals(NO_CONTENT.getStatusCode(), getStatus(httpPatch));
+
+        // Next, check headers for the binary; they should not have changed
+        final HttpHead head1 = new HttpHead(location);
+        try (final CloseableHttpResponse response = execute(head1)) {
+            binaryEtag2 = response.getFirstHeader("ETag").getValue();
+            binaryLastModed2 = response.getFirstHeader("Last-Modified").getValue();
+        }
+
+        assertEquals("ETags should be the same", binaryEtag1, binaryEtag2);
+        assertEquals("Last-Modified should be the same", binaryLastModed1, binaryLastModed2);
+
+        final HttpGet get6 = new HttpGet(location);
+        get6.addHeader("If-None-Match", binaryEtag1.toString());
```

Comment:
In principle, `get6` and `get7` are redundant. In practice, this is checking for the error that currently exists in the main codebase -- that the reported ETag doesn't match the ETag used internally.

Yes, I can change `binaryLastModed2` to `binaryLastModed1`. There is an earlier check to make sure those two values are the same, but that change might make it easier to follow.

I will remove the toString() in a subsequent commit.